tensorflow_addons/custom_ops/image/python/distort_image_ops_test.py

```diff
@@ -0,0 +1,347 @@
+            self._adjust_value_in_yiq_tf(x_np, scale)
+
+
+class AdjustSaturationInYiqTest(tf.test.TestCase):
+    def _adjust_saturation_in_yiq_tf(self, x_np, scale):
+        with self.cached_session(use_gpu=True):
+            x = tf.constant(x_np)
+            y = distort_image_ops.adjust_hsv_in_yiq(x, 0, scale, 1)
+        return y
+
+    def _adjust_saturation_in_yiq_np(self, x_np, scale):
+        """Adjust saturation using linear interpolation."""
+        rgb_weights = np.array([0.299, 0.587, 0.114])
+        gray = np.sum(x_np * rgb_weights, axis=-1, keepdims=True)
+        y_v = x_np * scale + gray * (1 - scale)
+        return y_v
+
+    @tf_test_util.run_all_in_graph_and_eager_modes
+    def test_adjust_random_saturation_in_yiq(self):
+        x_shapes = [
+            [2, 2, 3],
+            [4, 2, 3],
+            [2, 4, 3],
+            [2, 5, 3],
+            [1000, 1, 3],
+        ]
+        test_styles = [
+            "all_random",
+            "rg_same",
+            "rb_same",
+            "gb_same",
+            "rgb_same",
+        ]
+        with self.cached_session():
+            for x_shape in x_shapes:
+                for test_style in test_styles:
+                    x_np = np.random.rand(*x_shape) * 255.
+                    scale = np.random.rand() * 2.0 - 1.0
+                    if test_style == "all_random":
+                        pass
+                    elif test_style == "rg_same":
+                        x_np[..., 1] = x_np[..., 0]
+                    elif test_style == "rb_same":
+                        x_np[..., 2] = x_np[..., 0]
+                    elif test_style == "gb_same":
+                        x_np[..., 2] = x_np[..., 1]
+                    elif test_style == "rgb_same":
+                        x_np[..., 1] = x_np[..., 0]
+                        x_np[..., 2] = x_np[..., 0]
+                    else:
+                        raise AssertionError(
+                            "Invalid test style: %s" % (test_style))
+                    y_baseline = self._adjust_saturation_in_yiq_np(x_np, scale)
+                    y_tf = self._adjust_saturation_in_yiq_tf(x_np, scale)
+                    self.assertAllClose(y_tf, y_baseline, rtol=2e-4, atol=1e-4)
+
+    @tf_test_util.run_all_in_graph_and_eager_modes
+    def test_invalid_shapes(self):
+        x_np = np.random.rand(2, 3) * 255.
+        scale = np.random.rand() * 2.0 - 1.0
+        with self.assertRaisesRegexp(ValueError,
+                                     "Shape must be at least rank 3"):
+            self._adjust_saturation_in_yiq_tf(x_np, scale)
+        x_np = np.random.rand(4, 2, 4) * 255.
+        scale = np.random.rand() * 2.0 - 1.0
+        with self.assertRaisesOpError("input must have 3 channels "
+                                      "but instead has 4 channels"):
+            self._adjust_saturation_in_yiq_tf(x_np, scale)
+
+
+class AdjustHueInYiqBenchmark(tf.test.Benchmark):
```

Comment:
@karmel Hi, Karmel. Do you know what's the future for Benchmark class in tf 2.0?